### PR TITLE
Add custom headers to social token request [SDK-2080]

### PR DESCRIPTION
**Pull request by TLFilip**
### Changes

For security reasons we need an option to add custom headers for token request. Such a possibility exists for email/password authentication and refresh token. This PR enables it for social login by adding `WebAuthProvider.Builder` method `withHeaders`.

### Testing

To test it inspect the network traffic and notice custom headers in token request when executing OAuth2 flow.

- [x] This change adds unit test coverage

- [x] This change adds integration test coverage

- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [x] All existing and new tests complete without errors

### CODE REVIEW

**Comment by TLFilip** [comment]
Hey, let me know if there’s a chance to merge that PR and what is required to do so. Thanks!

**Comment by danielphillips** [comment]
@cocojoe  Hi Martin, as discussed a while ago, could you look into this PR please?

**Comment by lbalmaceda** [comment]
Apologies for missing this one. I might be able to take a look next week. I'm tracking this internally under `SDK-2080`.

File: auth0/src/main/java/com/auth0/android/provider/OAuthManager.java [hunk 1]
```diff
@@ -320,6 +327,14 @@ private void addPKCEParameters(Map<String, String> parameters, String redirectUr
         }
     }
 
+    private void addPKCEHeaders(@NonNull Map<String, String> httpHeaders) {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/provider/OAuthManager.java:330]: I'd merge this method into `addPKCEParameters`. Please, make the `createPKCE()` method take the headers in addition to the redirectUri. That way they are passed in the construction of the PKCE instance. That would as well remove the setter (pkce.setHeaders)

**Review by lbalmaceda — CHANGES_REQUESTED** [review]
Hey @TLFilip, sorry for the time it took us to review this. We're currently working on the next major and making some improvements (this included). We hope to deliver that soon.

That said, I think makes sense to have this in the current major before that new one is shipped. The PR looks good. I've left a change request to refactor a bit of the methods, especially given that most of the classes there are package private.

**Review by lbalmaceda — APPROVED** [review]

**Comment by lbalmaceda** [comment]
Thanks for the changes @TLFilip

**Pull request merged**